drivers/hmis/spec/system/hmis/form_hud_elements_spec.rb

```diff
@@ -0,0 +1,335 @@
+        expect(find('[data-testid="inputSum"]').text).to eq('$500.00')
+
+        expect do
+          save_ignoring_warnings
+        end.to change(e1.income_benefits, :count).by(1)
+
+        expect(e1.income_benefits.sole.total_monthly_income).to eq(500)
+      end
+
+      it 'does not collect SOAR question' do
+        assert_no_text 'Connection with SOAR'
+      end
+
+      context 'when SOAR should be collected' do
+        let!(:p1) { create :hmis_hud_project, data_source: ds1, organization: o1, project_type: 13, with_coc: true } # 13 = PH - Rapid Re-Housing
+        let!(:f1) { create :hmis_hud_funder, data_source: ds1, project: p1, funder: 33 } # VA SSVF
+
+        it 'collects SOAR question' do
+          assert_text 'Connection with SOAR'
+          fill_in 'Earned Income (i.e., employment income)', with: '200'
+          mui_select 'Yes', from: 'Connection with SOAR'
+
+          expect do
+            save_ignoring_warnings
+          end.to change(e1.income_benefits, :count).by(1)
+
+          expect(e1.income_benefits.sole.total_monthly_income).to eq(200)
+          expect(e1.income_benefits.sole.connection_with_soar).to eq(1)
+        end
+      end
+    end
+  end
+
+  describe 'benefits' do
+    let!(:p1) { create :hmis_hud_project, data_source: ds1, organization: o1, project_type: 3, with_coc: true }
+    let!(:f1) { create :hmis_hud_funder, data_source: ds1, project: p1 }
+
+    it 'hides health insurance table when No is selected' do
+      assert_text 'Covered by Health Insurance'
+      assert_text 'Select Insurance Provider(s)'
+      mui_radio_choose 'No', from: 'Covered by Health Insurance'
+      assert_no_text 'Select Insurance Provider(s)'
+
+      expect do
+        save_ignoring_warnings
+      end.to change(e1.income_benefits, :count).by(1)
+
+      expect(e1.income_benefits.sole.insurance_from_any_source).to eq(0)
+    end
+
+    it 'autoselects Yes when any health insurance provider is selected in table' do
+      find('[aria-label="Veteran\'s Health Administration (VHA)"]').check
+      find('[aria-label="Medicare"]').check
+      expect(mui_radio_value_for('Covered by Health Insurance')).to eq('YES')
+
+      expect do
+        save_ignoring_warnings
+      end.to change(e1.income_benefits, :count).by(1)
+
+      expect(e1.income_benefits.sole.insurance_from_any_source).to eq(1)
+      expect(e1.income_benefits.sole.vha_services).to eq(1)
+      expect(e1.income_benefits.sole.medicare).to eq(1)
+    end
+  end
+
+  describe 'disabilities' do
+    context 'with project type of ES and no funders' do
+      let!(:p1) { create :hmis_hud_project, data_source: ds1, organization: o1, user: u1, funders: [] }
+
+      it 'renders limited disability component, only collecting overall disabling condition' do
+        assert_text 'Disabling Condition'
+        assert_no_text 'The following questions help determine if there are additional housing services or benefits available for the client.'
```

Comment:
I think we mainly want to test that the disability-specific questions are not asked (like physical disability, substance use, etc). This text doesn't really matter / may change or be removed over time, which shouldn't impact this test. 

```suggestion
        assert_no_text 'Physical Disability'
```